java/splitter/splitter-lambda/src/main/java/sleeper/splitter/lambda/FindPartitionsToSplitLambda.java

```diff
@@ -63,10 +68,27 @@ public FindPartitionsToSplitLambda() {
         this.tablePropertiesProvider = new TablePropertiesProvider(instanceProperties, s3Client, dynamoDBClient);
     }
 
-    public void eventHandler(ScheduledEvent event, Context context) {
-        LOGGER.info("FindPartitionsToSplitLambda triggered at {}", event.getTime());
-        new FindPartitionsToSplit(instanceProperties, tablePropertiesProvider, stateStoreProvider,
-                new SqsSplitPartitionJobSender(tablePropertiesProvider, instanceProperties, sqsClient)::send).run();
-        LOGGER.info("FindPartitionsToSplitLambda finished at {}", LocalDateTime.now());
+    @Override
+    public Void handleRequest(SQSEvent event, Context context) {
+        Instant startTime = Instant.now();
+        LOGGER.info("Lambda triggered at {}", startTime);
+        event.getRecords().stream()
+                .map(SQSEvent.SQSMessage::getBody)
+                .peek(body -> LOGGER.info("Received message: {}", body))
+                .map(serDe::fromJson)
+                .forEach(this::run);
+
+        Instant finishTime = Instant.now();
+        LOGGER.info("Lambda finished at {} (ran for {})", finishTime, LoggedDuration.withFullOutput(startTime, finishTime));
+        return null;
+    }
+
+    private void run(InvokeForTableRequest tableRequest) {
+        tableRequest.getTableIds().stream()
+                .map(tablePropertiesProvider::getById)
+                .forEach(tableProperties -> {
+                    new FindPartitionsToSplit(instanceProperties, tablePropertiesProvider, stateStoreProvider,
+                            new SqsSplitPartitionJobSender(tablePropertiesProvider, instanceProperties, sqsClient)::send).run();
```

Comment:
This is calling FindPartitionsToSplit for all tables in the instance, rather than the ones in the batch.